src/EmailSynchronizer.php

```diff
@@ -226,4 +248,20 @@ private function parseDateTime(Message $parsedDocument)
 
         return $date;
     }
+
+    private function findEmailThreadId(string $inReplyTo): ?string
+    {
+        try {
+            $email = $this->emailRepository->getById($inReplyTo);
+        } catch (NotFound $e) {
+            // We did find the thread, let's move on
+            return null;
+        }
+        // If the email is not a thread root then we return the thread root ID
+        if ($email->getThreadId()) {
```

Comment:
Can the thread id be `null`? In line 193 it is set to the email id for root mails.

Reply:
Thanks, I have added more comments to explain.

Here the threadId can be null because we could be replying to the thread root (the thread root has a `null` getThreadId()).

Reply:
> (the thread root has a `null` getThreadId()).

Hmm, I don't think so, because of line 191-194:
https://github.com/mnapoli/externals/pull/120/files#diff-8133c15998ef143392dabbf7350cbf36R191-R194

So I think before this pull request the thread id for root mails was `null`, but with this pr they are equal to the email id (for root mails).

Reply:
Oh that is a good point you are correct. I checked in the database as well and it's not null.

So it should always return here. I don't feel comfortable changing the code (making the getter return a non-nullable `string` and making the database column not nullable) because the PR already changes many things, I'd rather limit the scope of the change ^^

But thanks! It's something I should improve later.